cocoa/src/toga_cocoa/fonts.py

```diff
@@ -65,40 +61,22 @@ def __init__(self, interface):
                             self.interface.size, family
                         )
                     )
+                    font = NSFont.systemFontOfSize(font_size)
 
-                # Convert the base font definition into a font with all the desired traits.
-                attributes_mask = 0
-                if self.interface.weight == BOLD:
-                    attributes_mask |= NSFontMask.Bold.value
-
-                if self.interface.style == ITALIC:
-                    attributes_mask |= NSFontMask.Italic.value
-                elif self.interface.style == SMALL_CAPS:
-                    attributes_mask |= NSFontMask.SmallCaps.value
-
-                if attributes_mask:
-                    font = NSFontManager.sharedFontManager.convertFont(
-                        font, toHaveTrait=attributes_mask
-                    )
-
-                full_name = "{family}{weight}{style}".format(
-                    family=family,
-                    weight=(" " + self.interface.weight.title())
-                    if self.interface.weight is not NORMAL
-                    else "",
-                    style=(" " + self.interface.style.title())
-                    if self.interface.style is not NORMAL
-                    else "",
+            # Convert the base font definition into a font with all the desired traits.
+            attributes_mask = 0
+            if self.interface.weight == BOLD:
+                attributes_mask |= NSFontMask.Bold.value
+            if self.interface.style == ITALIC:
+                attributes_mask |= NSFontMask.Italic.value
+            if self.interface.variant == SMALL_CAPS:
+                attributes_mask |= NSFontMask.SmallCaps.value
+            if attributes_mask:
+                font = NSFontManager.sharedFontManager.convertFont(
+                    font, toHaveTrait=attributes_mask
                 )
 
-                if font is None:
```

Comment:
Are you certain this error handling is no longer required? Core and system fonts will have all the weights/variants; but if a user-supplied font doesn't have a particular variant/weight, will this still succeed?

(And if it will - a comment explaining how/why would be worthwhile)

Reply:
It's no longer possible for `font` to be None here, because:
* When a custom font can't be found, I've added a line above to fall back on the system font. Previously in this case, the size, style and weight settings would all be ignored.
* According to the [`convertFont` documentation](https://developer.apple.com/documentation/appkit/nsfontmanager/1462345-convertfont) if it fails to find a font with the given traits, it returns the original font unchanged. I'll add a comment explaining this.

Also, note I've changed the indentation here so that the SYSTEM font uses the same bold/italic logic as everything else. Previously the italic setting would be ignored with a SYSTEM font.